k8s/kubevirt/kubevirt.go

```diff
@@ -0,0 +1,366 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/portworx/sched-ops/k8s/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	instance    Ops
+	once        sync.Once
+	vmiResource = schema.GroupVersionResource{Group: "kubevirt.io", Version: "v1", Resource: "virtualmachineinstances"}
+)
+
+type VirtualMachineInstance struct {
+	// RootDisk is the name of the volume that is used as a root disk in the VMI.
+	RootDisk string
+	// RootDiskPVC is the name of the PVC corresponding to the root disk.
+	RootDiskPVC string
+	// LiveMigratable indicates if VMI can be live migrated.
+	LiveMigratable bool
+}
+
+// Ops is an interface to perform generic Object operations
+type Ops interface {
+	// GetVirtualMachineInstance retrieves some info about the specified VMI
+	GetVirtualMachineInstance(ctx context.Context, namespace, name string) (*VirtualMachineInstance, error)
+
+	// SetConfig sets the config and resets the client
+	SetConfig(config *rest.Config)
+}
```

Comment:
To follow the same pattern as other schedops packages, can we create a separate interface VMI operations and add that interface in the parent `Ops` interface? Also the VMI related code can be moved to a separate file called `vmi.go` or `virtualmachineinstance.go`.

```
> kubevirt.go

type Ops interface {
    VirtualMachineInstanceOps
    SetConfig(...)
}
```
```
> virtualmachineinstance.go

type VirtualMachineInstanceOps {
    GetVirtualMachineInstance(...)
}
```